### PR TITLE
FEATURE: EGL-295 - hide the filter bar when all filters are hidden

### PR DESCRIPTION
**Pull request by khanhhuynh-gooddata**
JIRA: EGL-295

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16374/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17897/ : SUCCESS in 9m 39s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
 // (C) 2023 GoodData Corporation
 import { describe, it, expect, vi } from "vitest";
-import { getVisibilityIcon } from "../utils.js";
-import { DashboardDateFilterConfigModeValues } from "@gooddata/sdk-model";
+import { areAllFiltersHidden, getVisibilityIcon } from "../utils.js";
+import {
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts:4]: Can you please re-arrange the import to make it more easy to read? Please move lines 4 - 8 below line 2, and add an empty line to separate the import library and import files in the same project.
Ex:
```javascript
import { describe, it, expect, vi } from "vitest";
import {
DashboardAttributeFilterConfigModeValues,
    DashboardDateFilterConfigModeValues,
} from "@gooddata/sdk-model";
import cloneDeep from "lodash/cloneDeep.js";

import { areAllFiltersHidden, getVisibilityIcon } from "../utils.js";
import { filterBarAttributeFilterIndexes } from "./filterBar.fixture.js";
```

Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts:5]: updated!

File: libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts [hunk 2]
```diff
@@ -2,7 +2,17 @@
 import { IntlShape } from "react-intl";
 import { messages } from "../../locales.js";
 import { IFilterButtonCustomIcon, VisibilityMode } from "@gooddata/sdk-ui-filters";
-import { DashboardDateFilterConfigModeValues } from "@gooddata/sdk-model";
+import {
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts:5]: Please help to re-arrange the imports to make it more clear:
```javascript
import { IntlShape } from "react-intl";
import { IFilterButtonCustomIcon, VisibilityMode } from "@gooddata/sdk-ui-filters";
import {
    DashboardAttributeFilterConfigMode,
    DashboardAttributeFilterConfigModeValues,
    DashboardDateFilterConfigMode,
    DashboardDateFilterConfigModeValues,
} from "@gooddata/sdk-model";

import { messages } from "../../locales.js";
import {
    FilterBarAttributeItem,
    FilterBarAttributeItems,
    isFilterBarAttributeFilterPlaceholder,
} from "./filterBar/useFiltersWithAddedPlaceholder.js";
```

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts (hunk 2 above)
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts:5]: updated

File: libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts [hunk 3]
```diff
@@ -1,8 +1,20 @@
 // (C) 2023 GoodData Corporation
 import { IntlShape } from "react-intl";
-import { messages } from "../../locales.js";
+
```
Comment (vannguyengd) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts:3]: Please help to remove this empty line :) we don't need to separate the import libraries.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts (hunk 1 above)
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts:5]: Are there any rules for this?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts (hunk 1 above)
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts:4]: Anyway, updated!

File: libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts (hunk 3 above)
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/utils.ts]: done!

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16377/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17900/ : SUCCESS in 9m 21s

**Review by vannguyengd — APPROVED** [review]
LGTM!

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4264

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3377/ : SUCCESS in 10m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17901/ : SUCCESS in 9m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/986/ : SUCCESS in 9m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4540/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1257/ : FAILURE in 1m 30s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/895/ : SUCCESS in 8m 01s

**Pull request merged**